### DISCUSSION TITLE
Autocomplete functions signature

### DISCUSSION THREAD

**Discussion by acgc99** [godotengine/godot-proposals#8194] [Editor]
When you accept autocompletion for a built-in function, like `_set`, argument and returned types are not specified. Same thing when you connect signals. It would be nice to fill argument and returned types automatically to avoid errors.

![image](https://github.com/godotengine/godot-proposals/assets/96864661/8c34dcd4-dec9-4f6e-9460-c0e77be7a2fb)
 
`_set` autocompletion should return `func _set(property: StringName, value: Variant) -> bool:`.

**Comment by AThousandShips** [comment]
If you enable `text_editor/completion/add_type_hints` does it add these?

**Comment by acgc99** [comment]
I didn't know about that! Thanks!

**Comment by acgc99** [comment]
It there any reason why this is not checked by default? That kind of methods requiere specific signature.

**Reply by AThousandShips** [reply]
Because typing isn't enforced by default

As this is already supported I think this can be closed
